Moviebuff/ContentView.swift

```diff
@@ -0,0 +1,74 @@
+import SwiftUI
+struct Movie: Decodable {
+    let id: Int
+    let title: String
+    let posterpath: String
+    let releasedate:String
+    let overview:String
+    let rating:Int
+}
+struct DemoData: Codable {
+    let title: String
+    let description: String
+}
+struct ContentView: View {
+    @ObservedObject var imageLoader = ImageLoader()
+    @State var movies = [Movie]()
+    
+    var body: some View {
+        NavigationView {
+            List{
+                ForEach(movies, id: \.title) { movie in
+                    NavigationLink(destination: details(movie: movie)) {
+                        HStack {
+                            Image("Movie")
+                                .resizable()
+                                .frame(width: 50, height: 50)
+                                .clipShape(Circle())
+                                .overlay(Circle().stroke(Color.white, lineWidth: 4))
+                            Text(movie.title)
+                                .font(.headline)
+                        }.padding(7)
+                    }
+                }
+            }
+            .onAppear(perform: loadData)
+            .navigationTitle("Movies")
+            }
+        
+    }
+    
+    func loadData() {
+        guard let url = URL(string: "https://preetampothala.github.io/testapi/moviestrending.json") else {
+            print("Invalid URL")
+            return
+        }
+        let request = URLRequest(url: url)
+        print(request)
+ 
+        URLSession.shared.dataTask(with: request) {data, response, error in
+            if let data = data {
+                print(data)
+                //pull out the declaration of the decoder object
+                let decoder = JSONDecoder()
+                //decode directly to an array of User structs rather than a Response
+                if let decodedResponse = try?
```

Comment:
One reason you may be having trouble debugging this is that by using `try?`, you swallow up any errors that occur and you can't inspect the actual errors. You could refactor to use a `do`/`catch` block, where you can actually handle and help debug any decoding errors.

My refactoring of handling this callback could look as follows:

```Swift
        URLSession.shared.dataTask(with: request) {data, response, error in
            guard let data else {
                print("Fetch failed: \(error?.localizedDescription ?? "Unknown error")")
                return
            }

            // only if you want to print the data you're getting for debugging:
            print(String(data: data, encoding: .utf8) ?? "Could not convert data to string")

            do {
                //pull out the declaration of the decoder object
                let decoder = JSONDecoder()
                //decode directly to an array of User structs rather than a Response
                let decodedResponse = try decoder.decode([Movie].self, from: data)
                DispatchQueue.main.async {
                    //decodedResponse is now [Movie] rather than Response.Movie
                    self.movies = decodedResponse
                }
            } catch let decodeError {
                print("Error decoding.", decodeError)
            }
        }.resume()
```

Using this code, I can see that your fetch is successful, but you are failing in the decoding stage:


> Error decoding. dataCorrupted(Swift.DecodingError.Context(codingPath: [_JSONKey(stringValue: "Index 0", intValue: 0), CodingKeys(stringValue: "rating", intValue: nil)], debugDescription: "Parsed JSON number <7.1> does not fit in Int.", underlyingError: nil))

This shows me that there's a problem with the `rating` field, which is declared as an `Int`, but cannot be parsed as such. I changed that to a `Float`, and voila!

<img width="565" alt="Screenshot 2022-11-27 at 10 12 28 PM" src="https://user-images.githubusercontent.com/17274/204198921-67172b95-613a-4e99-ab70-07eec3566f0d.png">